src/post/post.repository.knex.ts

```diff
@@ -1,38 +1,151 @@
+            null,
+        );
+
+        const post = new Post(
+            row.id,
+            row.uuid,
+            author,
+            row.type,
+            row.audience,
+            row.title,
+            row.excerpt,
+            row.content,
+            new URL(row.url),
+            parseURL(row.image_url),
+            new Date(row.published_at),
+            row.like_count,
+            row.repost_count,
+            row.reply_count,
+            null,
+            null,
+            row.reading_time_minutes,
+            new URL(row.ap_id),
+        );
+
+        return post;
+    }
+
     async save(post: Post): Promise<void> {
-        if (post.isNew) {
-            const [id] = await this.db('posts').insert({
-                uuid: post.uuid,
-                type: post.type,
-                audience: post.audience,
-                author_id: post.author.id,
-                title: post.title,
-                excerpt: post.excerpt,
-                content: post.content,
-                url: post.url.href,
-                image_url: post.imageUrl?.href,
-                published_at: post.publishedAt,
-                in_reply_to: post.inReplyTo?.id,
-                thread_root: post.threadRoot?.id,
-                like_count: 0,
-                repost_count: 0,
-                reply_count: 0,
-                reading_time_minutes: post.readingTime,
-                ap_id: post.apId.href,
-            });
-            // Hacks? Mutate the Post so `isNew` returns false.
-            (post as any).id = id;
-        } else {
-            // Handle updates
+        const transaction = await this.db.transaction();
+
+        try {
+            const potentiallyNewLikes = post.getPotentiallyNewLikes();
+
+            if (post.isNew) {
+                const [id] = await transaction('posts').insert({
+                    uuid: post.uuid,
+                    type: post.type,
+                    audience: post.audience,
+                    author_id: post.author.id,
+                    title: post.title,
+                    excerpt: post.excerpt,
+                    content: post.content,
+                    url: post.url.href,
+                    image_url: post.imageUrl?.href,
+                    published_at: post.publishedAt,
+                    in_reply_to: post.inReplyTo?.id,
+                    thread_root: post.threadRoot?.id,
+                    like_count: potentiallyNewLikes.length,
+                    repost_count: 0,
+                    reply_count: 0,
+                    reading_time_minutes: post.readingTime,
+                    ap_id: post.apId.href,
+                });
+
+                if (potentiallyNewLikes.length > 0) {
+                    const likesToInsert = potentiallyNewLikes.map(
+                        (accountId) => ({
+                            account_id: accountId,
+                            post_id: id,
+                        }),
+                    );
```

Comment:
```
if (potentiallyNewLikes.length > 0) {
                    const likesToInsert = potentiallyNewLikes.map(
                        (accountId) => ({
                            account_id: accountId,
                            post_id: id,
                        }),
                    );
```

This part of the code is repeated in both if and else block, maybe we can have `likesToInsert` outside the if block along with `const potentiallyNewLikes = post.getPotentiallyNewLikes();`.

Reply:
They're slightly different because in one branch of the block we don't have access to the `id` until we write to the database, so we can't pull this out